### ISSUE TITLE
Implement multipart uploading for S3 files larger than 5GB 

### ISSUE THREAD

**Issue by denis-yuen** [ga4gh/dockstore#580]
### Desired behaviour

See https://docs.aws.amazon.com/AmazonS3/latest/dev/usingHLmpuJava.html
Currently, 5GB uploads result in 
```
com.amazonaws.services.s3.model.AmazonS3Exception: null (Service: Amazon S3; Status Code: 400; Error Code: EntityTooLarge; Request ID: tx00000000000000000fae4-0058886cf8-50570c-default), S3 Extended Request ID: 50570c-default-default
```

FYI @keiranmraine

### Environment (Browser or OS and Dockstore version)

1.1.2

**Comment by keiranmraine** [comment]
Untested (I need to set up a build environment yet).  Reading suggests this would replace the [S3 block](https://github.com/ga4gh/dockstore/blob/develop/dockstore-launcher/src/main/java/io/dockstore/common/FileProvisioning.java#L310-L334).

```java
        if (destPath.startsWith("s3://")) {
          AmazonS3 s3Client = FileProvisioning.getAmazonS3Client(config);
          TransferManager tx = TransferManagerBuilder.withS3Client(AmazonS3)
          String trimmedPath = destPath.replace("s3://", "");
          List<String> splitPathList = Lists.newArrayList(trimmedPath.split("/"));
          String bucketName = splitPathList.remove(0);

          Upload myUpload = tx.upload(bucketName, sourceFile.getName(), sourceFile);
 
          myUpload.addProgressListener(new ProgressListener() {
              ProgressPrinter printer = new ProgressPrinter();
              long runningTotal = 0;

              @Override
              public void progressChanged(ProgressEvent progressEvent) {
                  if (progressEvent.getEventType() == ProgressEventType.REQUEST_BYTE_TRANSFER_EVENT) {
                      runningTotal += progressEvent.getBytesTransferred();
                  }
                  printer.handleProgress(runningTotal, inputSize);
              }
          });
          // After the upload is complete, call shutdownNow to release the resources.
          tx.shutdownNow();
          System.out.println();
        } else {
```

This will allow up to 150GB

Will need to test the file size to set an appropriate chunk size as a maximum of 10,000 chunks are allowed and they default to 15MB.  Quite likely this will be exceeded for things like BAM files.  Looks like the following is what is needed:

```java
TransferManager tx = TransferManagerBuilder.withS3Client(AmazonS3).withMinimumUploadPartSize(XXXXX);
```

Doc for method [here](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/transfer/TransferManagerBuilder.html#withMinimumUploadPartSize-java.lang.Long-).

**Comment by denis-yuen** [comment]
Released as part of 1.1.5
